pkg/tagger/collectors/ecs_fargate_extract.go

```diff
@@ -30,7 +33,7 @@ func (c *ECSFargateCollector) parseMetadata(meta *v2.Task, parseAll bool) ([]*Ta
 	}
 	globalTags := config.Datadog.GetStringSlice("tags")
 
-	if parseAll {
+	doOnceOrchScope.Do(func() {
```

Comment:
we can add `sync.Once` to `ECSFargateCollector` as an embedded field and do the following
```suggestion
	c.Do(func() {
```

Reply:
Yeah so I am not opposed to adding this as a field in `ECSFargateCollector` however embedding it might be a bit too sneaky - after all we may have other resources that would require this one time initialization, but at a different point in time.

Reply:
As it is currently implemented, this tagger entry is very linked to the ECSFargateCollector, so if we are sure that we have only one instance of the ECSFargateCollector, we can include it in the `ECSFargateCollector` struct. like this it will be easier to unit-test in needed.

Reply:
Good point, sounds good, I will move it there.